lib/auth.js

```diff
@@ -48,6 +55,66 @@ Auth.prototype.bearer = function (bearer, sendImmediately) {
   }
 }
 
+
+Auth.prototype.negotiate = function(request, authHeader, sendImmediately) {
+  var self = this
+
+  if (kerberos === undefined) {
+    // Nothing to do if we don't have Kerberos
+    // or a current credential cache
+    return
+  }
+
+  self.hasAuth = true
+
+  if (!sendImmediately) {
```

Comment:
This should be:

```js
if (sendImmediately === false) {return}
```

By default `sendImmediately` should be `true`, including for `undefined`, that's how it's documented. And btw the test that I pushed won't work with `if (!sendImmediately)`